cmd/contour/cli.go

```diff
@@ -118,21 +140,242 @@ type stream interface {
 	Recv() (*envoy_discovery_v3.DiscoveryResponse, error)
 }
 
-func watchstream(st stream, typeURL string, resources []string) {
-	m := proto.TextMarshaler{
-		Compact:   false,
-		ExpandAny: true,
+func watchstream(log *logrus.Logger, st stream, typeURL string, resources []string, nack bool, NodeID string) {
+	m := protojson.MarshalOptions{
+		Multiline:     true,
+		Indent:        "  ",
+		UseProtoNames: true,
+	}
+
+	currentVersion := "0"
+
+	// Send the initial, non-ACK discovery request.
+	req := &envoy_discovery_v3.DiscoveryRequest{
+		TypeUrl:       typeURL,
+		ResourceNames: resources,
+		VersionInfo:   currentVersion,
+		Node: &corev3.Node{
+			Id: NodeID,
+		},
 	}
+	log.WithField("currentVersion", currentVersion).Info("Sending discover request")
+	fmt.Println(m.Format(req))
```

Comment:
I've used `fmt.Println` for printing out the marshaled Envoy messages to make them easier for a human to read. `contour cli` isn't really intended to be machine-consumable anyway, it's a troubleshooting tool for humans. 😄